components/organisation-unit-tree/src/organisation-unit-tree/organisation-unit-tree.js

```diff
@@ -43,10 +49,16 @@ const OrganisationUnitTree = ({
 
     useEffect(() => {
         // do not refetch on initial render
-        if (refetch && reloadId > 0) {
+        if (refetch && reloadId > 0 && reloadId !== prevReloadId) {
             refetch()
+            setPrevReloadId(reloadId)
         }
-    }, [reloadId, refetch])
+
+        return () =>
+            console.warn(
+                '@TODO: Why does this component unmount after a force reload?'
```

Comment:
An unmount would be if the second arg of `useEffect` was just `[]` - this callback was triggered as a result of `reloadId`/`prevReloadId` changing.

Reply:
I'm not sure I can follow.
Here's what I observed: The reload id does not change, so in theory this should work.
But for some reason the whole org unit tree component unmounts when the `forceReload` prop changes (or at least the returned callback gets called), which causes the hook to be run again..

Reply:
What I meant is that

```js
useEffect(() => {
  return () => {
    console.log("unmount")
  }
}, [])
```

would signal an unmount, but adding this to the component does not result in anything being logged to the console - so the `console.warn` in the `forceReaload` `useEffect` is being triggered due to a change in either `reloadId`, `prevReloadId` or `refetch`.

After investigating this a but further, I've discovered that the changes are being triggered by `refetch`, whose value changes due to `onComplete` and `onError` in `useRootOrgData` changing (see https://github.com/dhis2/app-runtime/blob/master/services/data/src/react/hooks/useQueryExecutor.ts#L44-L85 and https://github.com/dhis2/ui/blob/aad22793676c10e4dbcb103fcbd4ba930a04b7f4/components/organisation-unit-tree/src/organisation-unit-tree/use-root-org-data/use-root-org-data.js#L38-L52).

One way to prevent this issue is by wrapping `onComplete` and `onError` with `useCallback`:

```js
{
onComplete: useCallback(() => {
  // ...
}, [setState]),
onError: useCallback(() => {
  // ...
}, [setState]),
}
```

Reply:
We agreed that this will be solved in #681